### DISCUSSION TITLE
Not able to issue PUT command with query parameters from App service

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#234] [General]
Hi Team,

I am using REST device service to update specific record in the database in the end server. This requirement needs query parameters in the request.
This needs to be initiated from EdgeX app service.

But I am not able to find API to initiate PUT command with query parameters from the app service. 
Any help would be appreciated.

Thanks in advance,
Sudhamani.

**Comment by jiekechoo** [comment]
@sudhamani-hcl, I don't understand your point. The REST device service is for connect to south side device , and the app service is for upload OT data to cloud/enterprise, they are different.

**Reply by sudhamani-hcl** [reply]
Hi @jiekechoo ,

I have understood your point. But I am talking about on-demand request.

For example, I am able to read registers of devices on demand using API `IssueGetCommandByNameWithQueryParams` from app service. 
In the same way, I wanted to issue PUT command with query parameters to REST device service.

Is there any API to support this requirement? Any help would be appreciated.

Thanks,
Sudhamani

**Reply by jiekechoo** [reply]
Let me guess, do you want to send a command to the REST device by app service?

**Reply by sudhamani-hcl** [reply]
Yes @jiekechoo , sorry for the confusion. I wanted to send command with query parameters. Is there any API supporting this?

**Reply by zyqhpz** [reply]
There is a sample code for the PUT command in the camera-management app service example. Maybe it's something you look for.

https://github.com/edgexfoundry/edgex-examples/blob/77e035913dd1d869806c6913e9d730e7d3f78349/application-services/custom/camera-management/appcamera/commands.go#L165
